modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java

```diff
@@ -1111,6 +1113,126 @@ public void testEnableAccountEntryRestricted() throws Exception {
 		}
 	}
 
+	@Test
+	public void testEnableAccountEntryRestrictedForExternalStorageType()
+		throws Exception {
+
+		// Enabling account restriction to a custom salesforce object definition
+
+		ObjectDefinition objectDefinition1 =
+			_objectDefinitionLocalService.addCustomObjectDefinition(
+				TestPropsValues.getUserId(), false, false,
+				LocalizedMapUtil.getLocalizedMap("Able"), "Able", null, null,
+				LocalizedMapUtil.getLocalizedMap("Ables"),
+				ObjectDefinitionConstants.SCOPE_COMPANY,
+				ObjectDefinitionConstants.STORAGE_TYPE_SALESFORCE,
+				Collections.emptyList());
+
+		ObjectField objectField1 = ObjectFieldUtil.addCustomObjectField(
+			new TextObjectFieldBuilder(
+			).userId(
+				TestPropsValues.getUserId()
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				StringUtil.randomId()
+			).objectDefinitionId(
+				objectDefinition1.getObjectDefinitionId()
+			).required(
+				true
+			).build());
+
+		objectDefinition1 =
+			_objectDefinitionLocalService.
+				enableAccountEntryRestrictedForExternalStorageType(
+					objectField1);
+
+		Assert.assertTrue(objectDefinition1.isAccountEntryRestricted());
+		Assert.assertTrue(
+			objectDefinition1.getAccountEntryRestrictedObjectFieldId() > 0);
+		Assert.assertFalse(objectDefinition1.isSystem());
+		Assert.assertEquals(
+			objectDefinition1.getStorageType(),
+			ObjectDefinitionConstants.STORAGE_TYPE_SALESFORCE);
+
+		_objectDefinitionLocalService.deleteObjectDefinition(objectDefinition1);
+
+		// Enabling account restriction for default storage object definition
+		// using salesforce method
+
+		User user = TestPropsValues.getUser();
```

Comment:
Can you inline this when invoking `_objectDefinitionLocalService.addObjectDefinition`?
I think you can use TestPropsValues.getUserId() directly